File: core/src/ru/myitschool/dcrawler/entities/skills/action/ComplexEffectAction.java
package ru.myitschool.dcrawler.entities.skills.action;

import ru.myitschool.dcrawler.math.MathAction;
import ru.myitschool.dcrawler.entities.skills.Skill;

public abstract class ComplexEffectAction extends Action {

    public ComplexEffectAction(Skill skill) {
        super(skill);
    }

    @Override
    protected MathAction successDamage() {
        return null;
    }

    @Override
    protected MathAction critSuccessDamage(int standardDamage) {
        return null;
    }

    @Override
    protected MathAction failDamage() {
        return null;
    }

    @Override
    protected MathAction critFailDamage(int standardDamage) {
        return null;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/action/ComplexEffectAction.kt
package com.wolg_vlad.dcrawler.entities.skills.action

import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.math.MathAction

abstract class ComplexEffectAction(skill: Skill) : Action(skill) {
    override fun successDamage(): MathAction? {
        return null
    }

    override fun critSuccessDamage(standardDamage: Int): MathAction? {
        return null
    }

    override fun failDamage(): MathAction? {
        return null
    }

    override fun critFailDamage(standardDamage: Int): MathAction? {
        return null
    }
}